catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define*/
+define([
+        'icanhaz',
+        'backbone',
+        'marionette',
+        'underscore',
+        'jquery',
+        'q',
+        'wreqr',
+        'js/model/Node.js',
+        'js/view/NodeModal.view.js',
+        'js/view/DeleteModal.view.js',
+        'text!templates/registryPage.handlebars',
+        'text!templates/nodeList.handlebars',
+        'text!templates/nodeRow.handlebars'
+    ],
+    function (ich,Backbone,Marionette,_,$,Q,wreqr,Node,NodeModal,DeleteModal,registryPage, nodeList, nodeRow) {
+
+        var RegistryView = {};
+
+        ich.addTemplate('registryPage', registryPage);
+        ich.addTemplate('nodeList', nodeList);
+        ich.addTemplate('nodeRow', nodeRow);
+
+        RegistryView.RegistryPage = Marionette.Layout.extend({
+            template: 'registryPage',
+            events: {
+                'click .remove-node-link' : 'showRemoveNode',
+                'click .add-node-link' : 'showAddNode',
+                'click .refresh-button' : 'addDeleteNode'
+            },
+            initialize: function () {
+                this.listenTo(wreqr.vent, 'editNode', this.showEditNode);
+                this.listenTo(wreqr.vent, "deleteNodes", this.deleteNodes);
+                this.listenTo(wreqr.vent, "nodeUpdated", this.onRender);
+                this.listenTo(wreqr.vent, "nodeAdded", this.addDeleteNode);
+                this.listenTo(wreqr.vent, "nodeDeleted", this.addDeleteNode);
+
+                new RegistryView.ModalController({
+                    application: this
+                });
+                this.listenTo(this.model, 'add', this.render);
+                this.listenTo(this.model, 'remove', this.render);
+            },
+            regions: {
+                identityRegion: '#localIdentityNodeRegion',
+                additionalRegion: '#localAdditionalNodeRegion',
+                modalRegion: '#registry-modal'
+            },
+            onRender: function() {
+                var view = this;
+                view.secondaryNodes = [];
+                var identityNode;
+                _.each(view.model.models, function(model){
```

Comment:
Moved this logic out to two separate functions getSecondaryNodes and getIdentityNode